driver/kubernetes/kubernetes.go

```diff
@@ -258,24 +400,109 @@ func (k *Driver) Run(op *driver.Operation) (driver.OperationResult, error) {
 		defer k.deleteJob(job.ObjectMeta.Name)
 	}
 
-	// Return early for unit testing purposes (the fake k8s client implementation just
+	// Skip waiting for the job in unit tests (the fake k8s client implementation just
 	// hangs during watch because no events are ever created on the Job)
-	if k.skipJobStatusCheck {
-		return driver.OperationResult{}, nil
+	var opErr *multierror.Error
+	if !k.skipJobStatusCheck {
+		// Create a selector to detect the job just created
+		jobSelector := metav1.ListOptions{
+			LabelSelector: labels.Set(job.ObjectMeta.Labels).String(),
+			FieldSelector: newSingleFieldSelector("metadata.name", job.ObjectMeta.Name),
+		}
+
+		// Prevent detecting pods from prior jobs by adding the job name to the labels
+		podSelector := metav1.ListOptions{
+			LabelSelector: newSingleFieldSelector("job-name", job.ObjectMeta.Name),
+		}
+
+		err = k.watchJobStatusAndLogs(podSelector, jobSelector, op.Out)
+		if err != nil {
+			opErr = multierror.Append(opErr, errors.Wrapf(err, "job %s failed", job.Name))
+		}
+	}
+
+	opResult, err := k.fetchOutputs(op)
+	if err != nil {
+		opErr = multierror.Append(opErr, err)
+	}
+
+	return opResult, opErr.ErrorOrNil()
+}
+
+func (k *Driver) initJobVolumes(err error) error {
```

Comment:
I'm curious as to why this function takes an `error` as a parameter?  It doesn't seem like we inspect/use it prior to overriding its value.

Reply:
Oops, that was a mistake. 🤦‍♀️